### ISSUE TITLE
diagonal pattern for bar and stacked chart?

### ISSUE THREAD

**Issue by VijinR** [Abhinandan-Kushwaha/react-native-gifted-charts#123]
![Screenshot 2022-04-01 at 4 42 32 PM](https://user-images.githubusercontent.com/97605704/161252561-afb1c047-c367-40f6-8117-980eddd30a76.png)

Is it possible to give diagonal pattern like shown in above graph?

**Comment by jensonjohn001** [comment]
+1

**Comment by Abhinandan-Kushwaha** [comment]
Hi @VijinR 👋
Thanks for requesting the feature.

It's now available in version `1.0.15`🎉

This can be achieved using the props `barBackgroundPattern` and `patternId`
The `barBackgroundPattern` should be a Svg Pattern element. Here's an example-
```js
import {Path, Pattern} from 'react-native-svg';

const App = () => {
  const MyPattern = () => {
      return (
        <Pattern
          id="DiagonalLines"
          patternUnits="userSpaceOnUse"
          x="0"
          y="0"
          width={60}
          height={14}
          viewBox="0 0 40 20">
          <Path
            d={`M ${-1220} ${38} L ${465} ${0}`}
            stroke={'white'}
            strokeWidth={1.5}
            transform={{rotation: '-16'}}
          />
        </Pattern>
      );
    };
    
    const data=[
    {
      value: 110,
      barBackgroundPattern: MyPattern,
      patternId: 'DiagonalLines',
      frontColor: '#4444dd',
      spacing: 8,
    },
    {value: 130},
    {
      value: 120,
      barBackgroundPattern: MyPattern,
      patternId: 'DiagonalLines',
      frontColor: '#4444dd',
      spacing: 8,
    },
    {value: 160},
    {
      value: 190,
      barBackgroundPattern: MyPattern,
      patternId: 'DiagonalLines',
      frontColor: '#4444dd',
      spacing: 8,
    }]
    
    return <BarChart data={dtt} barWidth={40} spacing={30} />
 }
```
The output of the above code is-

![Screenshot 2022-04-02 at 4 27 58 PM](https://user-images.githubusercontent.com/20596944/161380130-5daa0a3e-c2f8-4950-beda-2838d197c479.png)

Note that we can pass any Svg pattern (not just lines). For example, if we use the following component as `barBackgroundPattern`-
```js
const TrianglePattern = () => {
  return (
    <Pattern
    id="DiagonalLines"
    patternUnits="userSpaceOnUse"
    x="0"
    y="0"
    width="20"
    height="20"
    viewBox="0 0 10 10"
  >
    <Path d="M 0 0 L 7 0 L 3.5 7 z" fill="red" stroke="blue" />
  </Pattern>
  );
};
```
Then the ouput will be-

![Screenshot 2022-04-02 at 4 30 45 PM](https://user-images.githubusercontent.com/20596944/161380199-04310ccc-a6ed-48e3-9648-1c4acdf2d878.png)

I hope it helps🤗
Cheers🍺

**Comment by VijinR** [comment]
@Abhinandan-Kushwaha thanks
